### PR TITLE
Fix ip address collection for awsvpc network mode

### PR DESCRIPTION
**Pull request by hkaj**
### What does this PR do?

Fix ip address collection for awsvpc network mode.

### Motivation

It didn't work, because we were decoding a container metadata payload as a task metadata payload (same url but with a `/task` at the end). I changed it so we now have a function for each of these payloads.

### CODE REVIEW

File: pkg/util/docker/containers.go [hunk 1]
```diff
@@ -348,3 +373,24 @@ func correctMissingIPs(addrs []containers.NetworkAddress, hostIPs []string) []co
 	}
 	return correctedAddrs
 }
+
+// cross IP addresses with ports
+func crossIPsWithPorts(addrs []containers.NetworkAddress, ports []nat.Port) []containers.NetworkAddress {
+	res := []containers.NetworkAddress{}
```
Comment (leeavital) [inline pkg/util/docker/containers.go]: nit:  this can be pre-sized  (feel free to ignore since both lists should always be quite small)

**Review by leeavital — DISMISSED** [review]

File: pkg/util/docker/containers.go [hunk 2]
```diff
@@ -348,3 +373,24 @@ func correctMissingIPs(addrs []containers.NetworkAddress, hostIPs []string) []co
...
 }
+
+// crossIPsWithPorts returns the product of a list of IP addresses and a list of ports
+func crossIPsWithPorts(addrs []containers.NetworkAddress, ports []nat.Port) []containers.NetworkAddress {
+	res := make([]containers.NetworkAddress, len(addrs)*len(ports))
+	c := 0
+
+	for _, addr := range addrs {
+		if len(ports) == 0 {
+			res = append(res, addr)
```
Comment (ahmed-mez) [inline pkg/util/docker/containers.go]: do we want to have network tuples with IPs without ports ?

Comment (hkaj) [reply pkg/util/docker/containers.go]: This should not happen, as exposedPorts is based on ports used by the container. However if it does, I think it's better to have a NetworkAddress w/o a port than nothing. I could be wrong though, what do you say @leeavital ? (i'm going to merge anyway to move forward, but we can revisit if that's not the case)

Comment (hkaj) [reply pkg/util/docker/containers.go]: cc @smehmood

**Review by ahmed-mez — APPROVED** [review]

**Pull request merged**